beaglebone/gps/src/__main__.py

```diff
@@ -10,23 +10,68 @@
 def main():
 
     gps = GPSData()
+    tempTimeStamp = 0
+    # 4 byte code for the register where cfg-rate is located
+    header1 = 181
+    header2 = 98
+    classId = 6
+    Id = 8
+    # message size (bytes)
+    payloadSize = 6
+    # frequency of data collection
+    milliseconds = 50
+    # number of measurements per solution
+    navRatio = 1
+    # time reference (UTC, GPS, etc)
+    timeRef = 1
+    # byteOffset
+    byteOffset = 0
+
+    # This block changes above constants int bytes or shorts
+    comBytes1 = struct.pack("<B", header1)
+    comBytes2 = struct.pack("<b", header2)
+    frequencyBytes = struct.pack(">h", milliseconds)
+    navRatioBytes = struct.pack(">h", navRatio)
+    timeRefBytes = struct.pack(">h", timeRef)
+
+    # Calculating checksum
+    Buffer = [classId, Id, payloadSize, frequencyBytes[0], frequencyBytes[1],
+              navRatioBytes[0], navRatioBytes[1], timeRefBytes[0],
+              timeRefBytes[1], byteOffset]
+    ck_a = 0
+    ck_b = 0
+    for i in range(10):
+        ck_a = ck_a + Buffer[i]
+        ck_b = ck_b + ck_a
+    # bitmasking (in equation in doc sheet)
+    ck_a = ck_a & 0xFF
+    ck_b = ck_b & 0xFF
+
+    ck_aBytes = struct.pack("<b", ck_a)
+    ck_bBytes = struct.pack("<B", ck_b)
+
     with serial.Serial(port="/dev/ttyS4", bytesize=serial.EIGHTBITS,
```

Comment:
As is now, the user is required to configure the UART manually from a `python3` session prior to running this program to use it. This should be handled in the startup of the program.